chain/epoch_manager/src/reward_calculator.rs

```diff
@@ -25,6 +26,22 @@ pub struct RewardCalculator {
     pub num_seconds_per_year: u64,
 }
 
+impl From<&GenesisConfig> for RewardCalculator {
+    fn from(config: &GenesisConfig) -> Self {
+        RewardCalculator {
+            max_inflation_rate: config.max_inflation_rate,
+            num_blocks_per_year: config.num_blocks_per_year,
+            epoch_length: config.epoch_length,
+            protocol_reward_rate: config.protocol_reward_rate,
+            protocol_treasury_account: config.protocol_treasury_account.to_string(),
+            online_max_threshold: config.online_max_threshold,
+            online_min_threshold: config.online_min_threshold,
+            #[cfg(feature = "protocol_feature_rectify_inflation")]
+            num_seconds_per_year: NUM_SECONDS_IN_A_YEAR,
+        }
+    }
+}
```

Comment:
This feels like a (minor) abuse of the language/library feature.  `From`  generally implies "lossless conversion", rather that projection, and often is associated with ownership transfer. In other words, From usually goes from a type to an equivalent or bigger type. In this case, we do the opposite and take subset of fields out of GenesisConfig

In this particular case, just an inherent `pub fn new`  would work?

Inherent methods are more discoverable, and more future-proof. Suppose in the future it turns out that we need something else, besides the GenessisConfig, to construct RewardCalculator. With `new`, we'd just add an extra argument. With `From`, we need to first refactor to using `new`. 

Finally, in general it's best to use the simplest tool that gets the job done. A function is simpler than a trait. 

To clarify, this isn't at all important in terms of actual, meaningful difference in this specific example. It's just an element of Rust style :)

Reply:
I see. Thanks for pointing this out!